### DISCUSSION TITLE
`pnpm link` an aliased module

### DISCUSSION THREAD

**Discussion by vjpr** [pnpm/pnpm#4693] [Ideas]
Say I want to replace a package with my fork like so:

```
{
  dependencies: {
    "next": "npm:@vjpr/next@^6.0.0-vjpr.1",
  }
}
```

My fork is renamed to `@vjpr/next` and publish to npm.

But now how can I `pnpm link` to it?

Running `pnpm link @vjpr/next` or `pnpm link next` does this:

```
/foo/node_modules/@vjpr/next -> /Users/Vaughan/nvm/versions/node/v8.9.0/pnpm-global/node_modules/@vjpr/next
```

But this doesn't help me...

**Comment by vjpr** [comment]
`pnpm link @vjpr/next` does add it to `node_modules`, so perhaps I could point next there manually like so:

```
ln -sf @vjpr/next node_modules/next
```

Then when I want to revert back I think I can run `pnpm dislink next`.

**Comment by zkochan** [comment]
Do you know how Yarn solves this?

**Comment by vjpr** [comment]
Couldn't find it from searching the issues. Would have to test it out.

**Comment by vjpr** [comment]
In terms of a solution, `pnpm link next` should show a warning that it is being aliased, and it should try to link the alias. There should be a flag that allows you to link to the original too in order to allow switching between linking your fork and the official repo.

**Comment by zkochan** [comment]
related #1364
